### PR TITLE
Refactored WiFi configuration window

### PR DESCRIPTION
**Pull request by ErrorErrorError**
Hey everyone!

I hope everyone is doing great! I just came with another update to make our lives a bit easier, at least in regards to duplicated classes for the same idea, WiFi windows!

This pull requests removes `JoinPopWindow`, `WiFiPopoverView`,  `WiFiPopupWindow`, and `PrefsViewWiFiInfoModal` in favor of a simplified class `WiFiConfigWindow`. The initial constructor without any arguments creates a `JoinPopWindow`. I hope this reduces the number of places to make changes, i.e. adding more security options. 

I also renamed the `edit` button from `PrefsSavedNetworksView` to `view` since you can only view the information as of now. Maybe in the future once more implementations of security are available, we can allow users to change their saved passwords, but as of now you can only view or removed saved networks.

### CODE REVIEW

**Comment by ErrorErrorError** [comment]
This should also fix part of https://github.com/OpenIntelWireless/HeliPort/issues/66 in regards to `JoinPopWindow is not resizeable when different encryptions are selected and everything overlays each other
` however I am not in Big Sur. If anyone can test and let me know that will be great!

File: HeliPort/Appearance/WiFiConfigWindow.swift [hunk 1]
```diff
@@ -0,0 +1,591 @@
...
+
+// MARK: Action Items
+
+extension WiFiConfigWindow {
+
+    @objc private func security(_ sender: Any?) {
+        switch securityPop.title {
+        case .none:
+            gridView.row(at: .usernameRow).isHidden = true
+            gridView.row(at: .passwordRow).isHidden = true
```
Comment (Goshin) [inline HeliPort/Appearance/WiFiConfigWindow.swift:421]: should clear up `passwdInputBox.stringValue` and `passwdSecureBox.stringValue.count` here.

Comment (ErrorErrorError) [reply HeliPort/Appearance/WiFiConfigWindow.swift:421]: After the switch case, it calls `resetInputBoxes()` and `controlJoinButton()` on lines 447 and 448, which clears out password and username and resets the `joinButton` state. However I can also just move those two methods in each switch case.

Comment (Goshin) [reply HeliPort/Appearance/WiFiConfigWindow.swift:421]: Oops, forget about it. I just missed that line.

**Comment by williambj1** [comment]
Thanks!!!

**Pull request merged**